modules/ivc_champva/app/models/ivc_champva/vha_10_10d.rb

```diff
@@ -38,9 +38,36 @@ def metadata
       }
     end
 
+    # rubocop:disable Layout/LineLength, Style/IfUnlessModifier
     def desired_stamps
-      [{ coords: [40, 105], text: data['statement_of_truth_signature'], page: 0 }]
+      return [] unless @data
+
+      stamps = [
+        { coords: [40, 105], text: @data['statement_of_truth_signature'], page: 0 }
+      ]
+
+      sponsor_is_deceased = @data.dig('veteran', 'sponsor_is_deceased')
+      veteran_country = @data.dig('veteran', 'address', 'country')
+      applicants = @data.fetch('applicants', [])
+      first_applicant_country = applicants.is_a?(Array) && !applicants.empty? ? applicants.first&.dig('applicant_address', 'country') : nil
+
+      stamps << { coords: [520, 470], text: first_applicant_country, page: 0 }
```

Comment:
Would it make sense to move this inside the loop rather than have it as a standalone?